### PR TITLE
[SPARK-15203][Deploy]The spark daemon shell script error, daemon process start successfully but script output fail message

### PR DESCRIPTION
**Pull request by WeichenXu123**
## What changes were proposed in this pull request?

fix the bug:
The spark daemon shell script error, daemon process start successfully but script output fail message
## How was this patch tested?

existing test.

### CODE REVIEW

**Comment by srowen** [comment]
Fix the title please?

File: sbin/spark-daemon.sh [hunk 1]
```diff
@@ -162,6 +162,16 @@ run_command() {
   esac
 
   echo "$newpid" > "$pid"
+  
+  #here add this loop to check the daemon process turn into java process. It takes at most 5 seconds.
```
Comment (srowen) [inline sbin/spark-daemon.sh]: This would be somewhat clearer as something like "Poll for up to 5 seconds for the java process to start" Otherwise LGTM

**Comment by WeichenXu123** [comment]
@srowen 
Modified as you expected.

**Comment by srowen** [comment]
Merged to master/2.0

**Pull request closed without merging**